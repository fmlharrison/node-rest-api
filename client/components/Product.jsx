import React from "react";
import PropTypes from "prop-types";

const Product = ({ info }) => {
  return (
    <div data-product-id={info.id}>
      <h4>{info.name}</h4>
      <p>{info.price}</p>
    </div>
  );
};

export default Product;

Product.propTypes = {
  info: PropTypes.shape({
    id: PropTypes.string,
    name: PropTypes.string,
    price: PropTypes.number
  }).isRequired
};
