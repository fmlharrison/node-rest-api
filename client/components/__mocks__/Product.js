export const product = {
  id: "12345",
  name: "Happy Meal",
  price: 5.99
};
