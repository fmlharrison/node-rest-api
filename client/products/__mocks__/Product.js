export const product = {
  id: "12345",
  name: "Happy Meal",
  price: 5.99
};

export const apiProductResponse = [
  {
    _id: "12345",
    name: "Happy Meal",
    price: 5.99
  }
];
