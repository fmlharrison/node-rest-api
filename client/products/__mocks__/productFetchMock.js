import { apiProductResponse } from "./Product";

// export default function apiGetMethod = jest.fn().mockImplementation(
//     new Promise((resolve, reject) => {
//         process.nextTick(
//             () => resolve(apiProductResponse);
//         );
//     });
// );
