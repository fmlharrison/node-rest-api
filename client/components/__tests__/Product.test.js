import React from "react";
import Product from "../Product.jsx";
import { product } from "../__mocks__/Product";

test("render the products list", () => {
  const wrapper = shallow(<Product info={product} />);
  expect(wrapper).toMatchSnapshot();
});

test("renders with the correct props", () => {
  const wrapper = mount(<Product info={product} />);
  expect(wrapper.props().info.id).toEqual("12345");
  expect(wrapper.props().info.name).toEqual("Happy Meal");
  expect(wrapper.props().info.price).toEqual(5.99);
});
