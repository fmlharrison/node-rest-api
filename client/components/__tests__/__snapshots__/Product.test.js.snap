// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render the products list 1`] = `
<div
  data-product-id="12345"
>
  <h4>
    Happy Meal
  </h4>
  <p>
    5.99
  </p>
</div>
`;
